geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/serialization/SimpleFeatureDecodings.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.locationtech.geomesa.feature.serialization
+
+import org.locationtech.geomesa.utils.cache.SoftThreadLocalCache
+import org.opengis.feature.simple.SimpleFeatureType
+
+import collection.JavaConversions._
+import CacheKeyGenerator.cacheKeyForSFT
+
+import scala.ref.SoftReference
+
+
+/** Provides access to the decodings for a [[SimpleFeatureType]].
+  *
+  */
+class SimpleFeatureDecodings[Reader](val datumReaders: AbstractReader[Reader], val sft: SimpleFeatureType) {
```

Comment:
oh yeah, the other thing you get with case classes is pattern matching unapply, e.g. 
obj match {
case SimpleFeatureDecodings(readers, sft) =>